### ISSUE TITLE
fabric8:debug allow suspend y/n to be optionally specified via a property

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8-maven-plugin#1003]
it would be nice to enable suspend mode when debugging so that the JVM pauses and waits for the debugger to connect before starting; for cases where you wanna debug a failing JVM

**Comment by nicolaferraro** [comment]
That would be very nice. E.g. if your app has a startup error when deployed, it's almost impossible to debug it without suspension.

**Comment by jstrachan** [comment]
agreed! it'd be a really nice handy way to debug startup errors. 

I think all we need is to just set the `suspend=y` on the debug system property thingy so shouldn't be crazy hard I don't think

**Comment by nicolaferraro** [comment]
Readiness checks may cause some problems. I don't know if you can port-forward before pod is ready, otherwise you may need to disable them.

**Comment by jstrachan** [comment]
ah good point; we may have to disable them for debugging

**Comment by nicolaferraro** [comment]
I can work on it. It needs some changes in the base image.

**Comment by metacosm** [comment]
Hi, any idea when this feature makes it into a release?

**Comment by nicolaferraro** [comment]
The code is already there, but the release involves changes the image (and we've been a bit disconnected in this period :sweat:)..

@rhuss can you generate a tag in [run-java-sh](https://github.com/fabric8io-images/run-java-sh), so we can use the debug properties in the image (that should be then re-generated and pushed to docker.io).

**Comment by metacosm** [comment]
@nicolaferraro doesn't seem like the code for the maven plugin has been merged into master yet, though…

**Comment by hrishin** [comment]
@nicolaferraro this issue is closed now, r8?

**Comment by nicolaferraro** [comment]
Sorry, haven't seen your previous comment and completely forgot...
I will have a second look at my branch tomorrow and merge on master.

This needs the updated image, I think @rhuss will be free to help on that at the end of this week.

**Comment by hrishin** [comment]
Thanks @nicolaferraro .
Hey @rhuss have we released the images with recent changes in `run.sh` script?

**Comment by nicolaferraro** [comment]
This forgotten change is finally there, with the new images :)
